paws_collector.js

```diff
@@ -495,6 +487,48 @@ class PawsCollector extends AlAwsCollector {
         });
     }
 
+    batchLogProcess(logs, privCollectorState, nextInvocationTimeout, callback) {
+        let collector = this;
+        return collector.processLog(logs, collector.pawsFormatLog.bind(collector), null,  async(err) => {
+           
+            if (err) {
+                /**
+                 * MS api return the logs but if build payload size > 10MB, and split logs array the in two part.
+                 * Call the processLog again to send Logmsgs to ingest. 
+                 */
+                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1)) {
+
+                    const half = Math.floor(logs.length / 2);
+                    const indexArray = [
+                        { start: 0, stop: half },
+                        { start: half, stop: logs.length }
+                    ];
+
+                    let promises = indexArray.map((logpart) => {
+                        return new Promise((resolve, reject) => {
+                            collector.batchLogProcess(logs.slice(logpart.start, logpart.stop), privCollectorState, nextInvocationTimeout, (err, res) => {
+                                if (err) {
+                                    reject(err);
+                                }
+                                else {
+                                    resolve(res);
+                                }
+                            });
+                        });
+                    });
+                    const results = await Promise.all(promises);
+                    return callback(null, results[results.length-1], nextInvocationTimeout);
```

Comment:
I think we need to return `privCollectorState ` instead of `results[results.length-1]`

Reply:
both are returning same value. I will use privCollectorState.